src/tyr/isochrone_serializer.cc

```diff
@@ -85,17 +85,43 @@ serializeIsochrones(const Api& request,
       }));
     }
   }
-  // Add original locations to the geojson
+  // Add input and snapped locations to the geojson
   if (show_locations) {
+    int idx = 0;
     for (const auto& location : request.options().locations()) {
-      features->emplace_back(
-          map({{"type", std::string("Feature")},
-               {"properties", map({})},
-               {"geometry", map({{"type", std::string("Point")},
-                                 {"coordinates", array({fp_t{location.ll().lng(), 6},
-                                                        fp_t{location.ll().lat(), 6}})}})}}));
+      // first add snapped points as MultiPoint feature per origin point
+      auto snapped_points_array = array({});
+      std::vector<std::vector<float>> snapped_points;
+      for (const auto& path_edge : location.path_edges()) {
+        const valhalla::LatLng& snapped_latlng = path_edge.ll();
+        const std::vector<float> current_coords{snapped_latlng.lng(), snapped_latlng.lat()};
+        // remove duplicates of path_edges in case the snapped object is a node
+        if (std::find(snapped_points.begin(), snapped_points.end(), current_coords) ==
+            snapped_points.end()) {
+          snapped_points.push_back(current_coords);
+          snapped_points_array->push_back(
+              array({fp_t{current_coords.at(0), 6}, fp_t{current_coords.at(1), 6}}));
+        }
+      };
```

Comment:
The `current_coords` vector is just so I can remove duplicates for the case when the snapped object is a node. Don't really like that, so any improvement suggestion is welcome.

Reply:
yep there is a common c++ pattern you can use here where you insert it into a "hash set" and when you do the insertion you check if it actually got inserted (meaning the set hadn't see this unique point before). here is some pseudo code:

```c++
std::unordered_set<midgard::PointLL> snaps;
for(auto loc : request.locs()){
  for(auto pe : loc.path_edges()) {
    if(snaps.insert(midgard::PointLL(pe.lng(), pe.lat())).second) {
      //add feature as we havent seen this snap point before
    }
  }
}
```

Reply:
Ok, thx, I changed accordingly.